[6.5][NET] Update NUI layouts (#1521)
Samsung/tizen-docs: docs/application/dotnet/guides/user-interface/nui/relative-layout.md

<style>
    .tabcontent img {
        border: 1px solid #555;
        max-width: 100% !important;
        max-height: 100%;
    }
</style>

> # Relative layout

Reviewer: We changed this before from Relative Layout to Relative layout, and now you want to restore that change ? We've got information that we should keep sentences grammar in headers.

        <RelativeLayout/>
    </View.Layout>

    <View x:Name="RedRect"
        BackgroundColor="1,0,0,1"
        WidthSpecification="100"
        HeightSpecification="100"/>

    <View x:Name="GreenRect"
        BackgroundColor="0,1,0,1" HeightSpecification="100"
        RelativeLayout.LeftTarget={x:Reference Name=RedRect}"
        RelativeLayout.RightTarget="{x:Reference Name=BlueRect}"
        RelativeLayout.FillHorizontal="true"
        RelativeLayout.LeftRelativeOffset="1.0"/>

    <View x:Name="BlueRect"
        BackgroundColor="0,0,1,1"
        WidthSpecification="100"
        HeightSpecification="100"
        RelativeLayout.RightRelativeOffset="1.0"
        RelativeLayout.LeftRelativeOffset="1.0"
        RelativeLayout.HorizontalAlignment="End"/>

    <View x:Name="YellowRect"
        BackgroundColor="1,1,0,1"
        HeightSpecification="100"
        RelativeLayout.LeftTarget="{x:Reference Name=Parent}"
        RelativeLayout.RightTarget="{x:Reference Name=BlueRect}"
        RelativeLayout.FillHorizontal="true"/>

> </View>

Reviewer 2: Could you clarify if this comment is fixed, I am unable to figure out?

[https://github.com/Samsung/tizen-docs/pull/1521#discussion_r742537933](https://github.com/Samsung/tizen-docs/pull/1521#discussion_r742537933)

    </div>
</div>

### Red view

The following code shows how to create redView with bounded by offsets.

![RelativeLayoutExampleRedView](./media/relative_layout_example_red_view.png)

<div id="TabSection2">
    <div class="sampletab " id="ProjectCreateTab">
        <button id="RelativeLayout-RedView-CSharp" class="tablinks " onclick="openTabSection(event, 'RelativeLayout-RedView-CSharp', 'TabSection2') ">C#</button>
        <button id="RelativeLayout-RedView-Xaml" class="tablinks " onclick="openTabSection(event, 'RelativeLayout-RedView-Xaml', 'TabSection2') ">Xaml</button>
    </div>
    <div id="RelativeLayout-RedView-CSharp" class="tabcontent">
        <table>
            <tbody>
                <tr>
<span style="display:block">

- `RelativeLayout.SetLeftRelativeOffset(redView, 0.0f)` - view's left is at the parent's left.
- `RelativeLayout.SetRightRelativeOffset(redView, 0.1f)` - view's right is at the position of 0.1 width from the parent's left.
- `RelativeLayout.SetTopRelativeOffset(redView, 0.0f)` - view's top is at the parent's top.
- `RelativeLayout.SetBottomRelativeOffset(redView, 0.1f)` - view's bottom bound is at the position of 0.1 width from the parent's top.

> - `RelativeLayout.SetFillHorizontal=(redView, true)` and `SetRelativeLayout.FillVertical(redView, true)` - Since the view does not have its Width and Height,the view should be filled the relative position(left, right, top, bottom).

Reviewer 3: Changing `RelativeLayout.SetFillHorizontal=(redView, true)` to `RelativeLayout.SetFillHorizontal(redView, true)` is missed.
Please remove the equal sign.

```csharp
View redView = new View
{
    BackgroundColor = Color.Red
};

RelativeLayout.SetLeftRelativeOffset(blueView, 0.0f);
RelativeLayout.SetRightRelativeOffset(blueView, 0.1f);
RelativeLayout.SetTopRelativeOffset(blueView, 0.0f);
RelativeLayout.SetBottomRelativeOffset(blueView, 0.1f);

RelativeLayout.SetFillHorizontal(redView, true);
RelativeLayout.SetFillVertical(redView, true);

relativeLayoutView.Add(redView);
```

</span>
                </tr>
            </tbody>
        </table>
    </div>
    <div id="RelativeLayout-RedView-Xaml" class="tabcontent">
        <table>
            <tbody>
                <tr>
<span style="display:block">

- `RelativeLayout.LeftRelativeOffset="0.0"` - view's left is at the parent's left.
- `RelativeLayout.RightRelativeOffset="0.1"` - view's right is at the position of 0.1 width from the parent's left.
- `RelativeLayout.TopRelativeOffset="0.0"` - view's top is at the parent's top.
- `RelativeLayout.BottomRelativeOffset="0.1"` - view's bottom bound is at the position of 0.1 width from the parent's top.

> - `RelativeLayout.FillHorizontal="true"` and `RelativeLayout.FillVertical="true"` - Since the view does not have its Width and Height,the view should be filled the relative position(left, right, top, bottom).

Reviewer 3: ```suggestion
- `RelativeLayout.FillHorizontal="true"` and `RelativeLayout.FillVertical="true"` - Since the view does not have its Width and Height, the view should be filled the relative position(left, right, top, bottom).
```

Reviewer 3: This comment is not applied. Would you please check?

![RelativeLayoutExampleGreenView](./media/relative_layout_example_green_view.png)

<div id="TabSection5">
    <div class="sampletab " id="ProjectCreateTab">
        <button id="RelativeLayout-GreenView-CSharp" class="tablinks " onclick="openTabSection(event, 'RelativeLayout-GreenView-CSharp', 'TabSection5') ">C#</button>
        <button id="RelativeLayout-GreenView-Xaml" class="tablinks " onclick="openTabSection(event, 'RelativeLayout-GreenView-Xaml', 'TabSection5') ">Xaml</button>
    </div>
    <div id="RelativeLayout-GreenView-CSharp" class="tabcontent">
        <table>
            <tbody>
                <tr>
<span style="display:block">

- View's targets are set to the `yellowView`.
- `RelativeLayout.SetLeftRelativeOffset(greenView, 0.0f)` - view's left is at the yellowView's left.
- `RelativeLayout.SetRightRelativeOffset(greenView, 0.2f)` - view's right is at the position of 0.2 width from the yellowView's left.
- `RelativeLayout.SetTopRelativeOffset(greenView, 0.0f)` - view's top is at the yellowView's top.
- `RelativeLayout.SetBottomRelativeOffset(greenView, 0.2f)` - view's bottom is at the position of 0.2 width from the yellowView's left.

> - `RelativeLayout.SetFillHorizontal=(greenView, true)` and `SetRelativeLayout.FillVertical(greenView, true)` - Since the view does not have its Width and Height,the view should be filled the relative position(left, right, top, bottom).

Reviewer 3: 1. Changing `RelativeLayout.SetFillHorizontal=(greenView, true)` to `RelativeLayout.SetFillHorizontal(greenView, true)` is missed. 
Please remove the equal sign.
2. SetRelativeLayout.FillVertical(greenView, true) is changed to **Set**RelativeLayout.**Set**FillVertical(greenView, true),
 remaining `Set` at the front. 
Would you please check and update with the actual function name?

```csharp
View greenView = new View
{
    BackgroundColor = Color.Green
};

RelativeLayout.SetLeftTarget(greenView, yellowView);
RelativeLayout.SetRightTarget(greenView, yellowView);
RelativeLayout.SetTopTarget(greenView, yellowView);
RelativeLayout.SetBottomTarget(greenView, yellowView);

RelativeLayout.SetLeftRelativeOffset(greenView, 0.0f);
RelativeLayout.SetRightRelativeOffset(greenView, 0.2f);
RelativeLayout.SetTopRelativeOffset(greenView, 0.0f);
RelativeLayout.SetBottomRelativeOffset(greenView, 0.2f);

RelativeLayout.SetFillHorizontal(greenView, true);
RelativeLayout.SetFillVertical(greenView, true);
relativeLayoutView.Add(greenView);
```

- View's targets are set to the `yellowView`.
- `RelativeLayout.LeftRelativeOffset="0.0"` - view's left is at the yellowView's left.
- `RelativeLayout.RightRelativeOffset="0.2"` - view's right is at the position of 0.2 width from the yellowView's left.
- `RelativeLayout.TopRelativeOffset="0.0"` - view's top is at the yellowView's top.
- `RelativeLayout.BottomRelativeOffset="0.2"` - view's bottom is at the position of 0.2 width from the yellowView's left.

> - `RelativeLayout.FillHorizontal="true"` and `RelativeLayout.FillVertical="true"` - Since the view does not have its Width and Height,the view should be filled the relative position(left, right, top, bottom).

Reviewer 3: ```suggestion
- `RelativeLayout.FillHorizontal="true"` and `RelativeLayout.FillVertical="true"` - Since the view does not have its Width and Height, the view should be filled the relative position(left, right, top, bottom).
```

```

</span>
                </tr>
            </tbody>
        </table>
    </div>
</div>

### Black view

The following code shows how to create black view related to yellowView and greenView.

![RelativeLayoutExampleBlackView](./media/relative_layout_example_black_view.png)

<div id="TabSection6">
    <div class="sampletab " id="ProjectCreateTab">
        <button id="RelativeLayout-BlackView-CSharp" class="tablinks " onclick="openTabSection(event, 'RelativeLayout-BlackView-CSharp', 'TabSection6') ">C#</button>
        <button id="RelativeLayout-BlackView-Xaml" class="tablinks " onclick="openTabSection(event, 'RelativeLayout-BlackView-Xaml', 'TabSection6') ">Xaml</button>
    </div>
    <div id="RelativeLayout-BlackView-CSharp" class="tabcontent">
        <table>
            <tbody>
                <tr>
<span style="display:block">

- View's left and top targets are set to `greenView`.
- View's right and bottom targets are set to `yellowView`.
- `RelativeLayout.SetLeftRelativeOffset(blackView, 1.0f)` - view's left is on the greenView's right.
- `RelativeLayout.SetRightRelativeOffset(blackView, 1.0f)` - view's right is on the yellowView's right.

> - `RelativeLayout.SetTopRelativeOffset=(blackView, 1.0f)` - view's top is on the greenView's bottom.

Reviewer 3: ```suggestion
- `RelativeLayout.SetTopRelativeOffset(blackView, 1.0f)` - view's top is on the greenView's bottom.
```

Author: Done

> - `RelativeLayout.SetBottomRelativeOffset=(blackView, 1.0f)` - view's bottom is on the yellowView's bottom.

Reviewer 3: ```suggestion
- `RelativeLayout.SetBottomRelativeOffset(blackView, 1.0f)` - view's bottom is on the yellowView's bottom.
```

Author: Done

> - `RelativeLayout.SetSetFillHorizontal=(blackView, true)` and `SetRelativeLayout.FillVertical(blackView, true)` - Since the view does not have its Width and Height,the view should be filled the relative position(left, right, top, bottom).

Reviewer 3: ```suggestion
- `RelativeLayout.SetFillHorizontal(blackView, true)` and `RelativeLayout.SetFillVertical(blackView, true)` - Since the view does not have its Width and Height, the view should be filled the relative position(left, right, top, bottom).
```

```csharp
View blackView = new View
{
    BackgroundColor = Color.Black,
    Margin = new Extents(50, 50, 50, 50)
};

> RelativeLayout.SetLeftTarget(blackView, yellowView);

Reviewer 3: Question. Shouldn't it be greenView, not yellowView?

RelativeLayout.SetLeftTarget(blackView, **yellowView**);

Author: Done

RelativeLayout.SetRightTarget(blackView, yellowView);

> RelativeLayout.SetTopTarget(blackView, yellowView);

Reviewer 3: Question. Shouldn't it be greenView, not yellowView?

RelativeLayout.SetTopTarget(blackView, **yellowView**);

Author: Done

    <div id="RelativeLayout-RedView-Xaml" class="tabcontent">
        <table>
            <tbody>
                <tr>
<span style="display:block">

- `RelativeLayout.LeftRelativeOffset="0.0"` - view's left is at the parent's left.
- `RelativeLayout.RightRelativeOffset="0.1"` - view's right is at the position of 0.1 width from the parent's left.
- `RelativeLayout.TopRelativeOffset="0.0"` - view's top is at the parent's top.
- `RelativeLayout.BottomRelativeOffset="0.1"` - view's bottom bound is at the position of 0.1 width from the parent's top.
- `RelativeLayout.FillHorizontal="true"` and `RelativeLayout.FillVertical="true"` - Since the view does not have its Width and Height, the view should be filled with the relative position (left, right, top, and bottom).

```xaml
<View x:name="redView"
    BackgroundColor="Red"
    RelativeLayout.LeftRelativeOffset="0.0"
    RelativeLayout.RightRelativeOffset="0.1"
    RelativeLayout.TopRelativeOffset="0.0"
    RelativeLayout.BottomRelativeOffset="0.1"
    RelativeLayout.FillHorizontal="True"
    RelativeLayout.FillVertical="True"/>
```

</span>
                </tr>
            </tbody>
        </table>
    </div>
</div>

### Blue view

> The following code shows how to position the blue view with predefined size and change its position begins with `HorizontalAlignment`.

Reviewer 4: ```suggestion
The following code shows how to position the blue view with predefined size and change its position beginning with `HorizontalAlignment`.
```

![RelativeLayoutExampleBlueView](./media/relative_layout_example_blue_view.png)

<div id="TabSection3">
    <div class="sampletab " id="ProjectCreateTab">
        <button id="RelativeLayout-BlueView-CSharp" class="tablinks " onclick="openTabSection(event, 'RelativeLayout-BlueView-CSharp', 'TabSection3') ">C#</button>
        <button id="RelativeLayout-BlueView-Xaml" class="tablinks " onclick="openTabSection(event, 'RelativeLayout-BlueView-Xaml', 'TabSection3') ">Xaml</button>
    </div>
    <div id="RelativeLayout-BlueView-CSharp" class="tabcontent">
        <table>
            <tbody>
                <tr>
<span style="display:block">

- `WidthSpecification` and `HeightSpecification` - set the size of the view.

> - `RelativeLayout.SetLeftRelativeOffset(blueView, 1.0f)` - view's left is on the parent's left.

Reviewer: It is confusing for developers too, but SetLeftRelativeOffset() is related with the left side of the parent

> - `RelativeLayout.SetLeftRelativeOffset(blueView, 1.0f)` - view's left is on the parent's left.

Reviewer 4: "view’s left is on the parent’s right" is correct but shouldn't it be a certain width away from the parent's right?
I'm basing my comment on the "Red View" description.

> - `RelativeLayout.SetLeftRelativeOffset(blueView, 1.0f)` - view's left is on the parent's left.

Reviewer 3: Hello @asm-foysal 

I thought the left edge of blueView is aligned to the right edge of the parent,
and is away as much as the width of the blue view.
(The right edge of blueView is also aligned to the right edge of the parent)

```
    WidthSpecification = 200,
```

@mihashco If I misunderstood something, would you please let me know?

> - `RelativeLayout.SetLeftRelativeOffset(blueView, 1.0f)` - view's left is on the parent's left.

Reviewer: @mijin85cho, You're right. 
@asm-foysal, shouldn't it be a certain width away from the parent's right? - if relative ofset is set to 0.0 it means it is aligned to the edge.

    RelativeLayout.VerticalAlignment="Center"/>
```

</span>
                </tr>
            </tbody>
        </table>
    </div>
</div>

### Green view

The following code shows how to create the green view to relate to yellowView for every target:
![RelativeLayoutExampleGreenView](./media/relative_layout_example_green_view.png)

<div id="TabSection5">
    <div class="sampletab " id="ProjectCreateTab">
        <button id="RelativeLayout-GreenView-CSharp" class="tablinks " onclick="openTabSection(event, 'RelativeLayout-GreenView-CSharp', 'TabSection5') ">C#</button>
        <button id="RelativeLayout-GreenView-Xaml" class="tablinks " onclick="openTabSection(event, 'RelativeLayout-GreenView-Xaml', 'TabSection5') ">Xaml</button>
    </div>
    <div id="RelativeLayout-GreenView-CSharp" class="tabcontent">
        <table>
            <tbody>
                <tr>
<span style="display:block">

- View's targets are set to the `yellowView`.
- `RelativeLayout.SetLeftRelativeOffset(greenView, 0.0f)` - view's left is at the yellowView's left.
- `RelativeLayout.SetRightRelativeOffset(greenView, 0.2f)` - view's right is at the position of 0.2 width from the yellowView's left.

> - `RelativeLayout.SetTopRelative Offset(greenView, 0.0f)` - view's top is at the yellowView's top.

Reviewer 3: Would you please check if the space in the middle is needed?

```suggestion
- `RelativeLayout.SetTopRelativeOffset(greenView, 0.0f)` - view's top is at the yellowView's top.
```

Reviewer: Fixed

 |:---:|:---:|:---:|
 |![RelativeAlignStart](./media/relativeLayoutAlignStart.png)|![RelativeAlignCenter](./media/relativeLayoutAlignCenter.png)|![RelativeAlignEnd](./media/relativeLayoutAlignEnd.png)|

> The alignment property allows to change fixed size view's position begins when all targets on axis (horizontal or vertical) are set to the same value.

Reviewer 4: I concur with Sagar, the sentence is difficult to understand. Please rephrase the sentence or explain what you are trying to convey here so I may provide a suggestion.

|`HorizontalAlignment`: Start |`HorizontalAlignment`: Center|`HorizontalAlignment`: End|
| :---: | :---: | :---: |
| ![RelativeImagePositionBeginsStart](./media/relativeLayoutAlignPositionBeginStart.png) | ![RelativeImagePositionBeginsCenter](./media/relativeLayoutAlignPositionBeginCenter.png) | ![RelativeImagePositionBeginsEnd](./media/relativeLayoutAlignPositionBeginEnd.png) |

## Fill

 The `FillHorizontal` and the `FillVertical` properties are specified by boolean values. These boolean values specify whether or not to fill the layout space. The default is `false` for both the axes:
 |`FillHorizontal`: false|`FillHorizontal`: true|
 |:---:|:---:|
 |![RelativeAlignCenter](./media/relativeLayoutAlignCenter.png)|![RelativeFill](./media/relativeLayoutFill.png)|

> [!NOTE]

> >To fill control with the predefined size is needed to change property `WidthSpecification`/`HeightSpecification` to value `WrapContent`. Otherwise the control doesn't fill properly the space created by offsets.

Reviewer 4: Same here, the sentence is difficult to understand. Please rephrase the sentence or explain what you are trying to convey here so I may provide a suggestion.

    RelativeLayout.RightRelativeOffset="0.1"
    RelativeLayout.TopRelativeOffset="0.0"
    RelativeLayout.BottomRelativeOffset="0.1"
    RelativeLayout.FillHorizontal="True"
    RelativeLayout.FillVertical="True"/>
```

</span>
                </tr>
            </tbody>
        </table>
    </div>
</div>

### Blue view

The following code shows how to position the blue view with predefined size and change its position begins with `HorizontalAlignment`.

![RelativeLayoutExampleBlueView](./media/relative_layout_example_blue_view.png)

<div id="TabSection3">
    <div class="sampletab " id="ProjectCreateTab">
        <button id="RelativeLayout-BlueView-CSharp" class="tablinks " onclick="openTabSection(event, 'RelativeLayout-BlueView-CSharp', 'TabSection3') ">C#</button>
        <button id="RelativeLayout-BlueView-Xaml" class="tablinks " onclick="openTabSection(event, 'RelativeLayout-BlueView-Xaml', 'TabSection3') ">Xaml</button>
    </div>
    <div id="RelativeLayout-BlueView-CSharp" class="tabcontent">
        <table>
            <tbody>
                <tr>
<span style="display:block">

- `WidthSpecification` and `HeightSpecification` - set the size of the view.

> - `RelativeLayout.SetLeftRelativeOffset(blueView, 1.0f)` - view's left is on the parent's right.

Reviewer 4: I think Sagar's suggested change is correct, can you please check and confirm?

> - `RelativeLayout.SetLeftRelativeOffset(blueView, 1.0f)` - view's left is on the parent's right.

Reviewer: done
